⭐Add a prod build make job.

preslavgerchev opened the pull request:
I noticed during testing that the mondoo version and build are empty:
`==> amazon-ebs: Running mondoo (Version: , Build: )`

I added a `build/prod` make job that correctly assigns the values after which this gets fixed.

`==> amazon-ebs: Running Mondoo packer provisioner (Version: v0.2.1, Build: af68414)`

I am not sure on the release process and what needs to be changed there, could use some help if this PR gets approved

preslavgerchev commented on provisioner/provisioner.go:
```diff
@@ -157,7 +157,7 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 
 func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.Communicator, generatedData map[string]interface{}) error {
-	ui.Say("Running mondoo (Version: " + version.Version + ", Build: " + version.Build + ")")
+	ui.Say("Running Mondoo packer provisioner (Version: " + version.Version + ", Build: " + version.Build + ")")
```
I am proposing this in place of just `mondoo` as it was slightly confusing with the mondoo client reporting its status directly underneath, e.g.
```
==> amazon-ebs: Running mondoo (Version: v0.2.1, Build: 7d7b5e1)
==> amazon-ebs: Executing Mondoo: <redacted>
    amazon-ebs: → Mondoo 6.3.0 <redacted>
```

vjeffrey replied:
that's a great improvement, thank you!

vjeffrey commented:
re: release, see https://github.com/mondoohq/packer-plugin-mondoo/actions/workflows/release.yml && https://github.com/mondoohq/packer-plugin-mondoo/blob/main/.github/workflows/release.yml

chris-rock commented:
@preslavgerchev The released binary should include the versions, they are not built with the make task but instead use the goreleaser task. If we try to get some useful data into a local build, we need to update the `make/dev` task to include version and build. If we do this we should also attach a `+dev` to the last tag. So that the version would be `0.1.2+dev`

chris-rock reviewed (changes requested):
we should not add a `build/prod` since we do this via goreleaser. if we want to include version data for the dev build, we need to update the `make build/dev` task instead.

preslavgerchev commented:
> we should not add a `build/prod` since we do this via goreleaser. if we want to include version data for the dev build, we need to update the `make build/dev` task instead.

Thank you, figured out how we do it. I removed the `build/prod` task and updated the ldflags in the .goreleaser.yml file instead. Didn't update `build/dev` as I am also fine with empty data there since it's intended for developing

chris-rock reviewed (approved):
Thank you for the improvement @preslavgerchev
